File: Trabalhos Praticos/tp01/entrada e saida/TP01Q07 - Is em C/Isthing.c
#include <stdio.h>
#include <string.h>

#define bool short
#define true 1
#define false 0

bool IsVogal(char str[])
{
    int tam = strlen(str);
    int tamchecker = strlen(str);
    // printf("%d", tamchecker);
    for (int i = 0; i < tam; i++)
    {
        if ((str[i] == 'a' || str[i] == 'A') ||
            (str[i] == 'e' || str[i] == 'E') ||
            (str[i] == 'i' || str[i] == 'I') ||
            (str[i] == 'o' || str[i] == 'O') ||
            (str[i] == 'u' || str[i] == 'U') ||
            str[i] == ' ')
        {
            tamchecker -= 1;
        }
        /*
        if (tamchecker == 0)
        {
            return true;
        }
        */
    }
    return !tamchecker;
    // return false;
}

bool IsConsoante(char str[])
{
    int tam = strlen(str);
    int tamchecker = 0;
    // printf("%d", tamchecker);
    for (int i = 0; i < tam; i++)
    {
        // Verifica se é uma letra do alfabeto
        if ((str[i] >= 'a' && str[i] <= 'z') || (str[i] >= 'A' && str[i] <= 'Z'))
        {
            // Verifica se é uma vogal
            if (str[i] == 'a' || str[i] == 'A' ||
                str[i] == 'e' || str[i] == 'E' ||
                str[i] == 'i' || str[i] == 'I' ||
                str[i] == 'o' || str[i] == 'O' ||
                str[i] == 'u' || str[i] == 'U')
            {
                tamchecker--; // Decrementa  se for uma vogal
            }
        }
        else
        {
            tamchecker--; // Decrementa se não for uma letra do alfabeto
        }
    }
    // printf("%d", tamchecker);
    return !tamchecker;
}

bool IsInteiro(char str[])
{
    int tam = strlen(str);
    int tamchecker = strlen(str);
    for (int i = 0; i < tam; i++)
    {
        if (str[i] >= '0' && str[i] <= '9')
        {
            tamchecker -= 1;
        }
    }
    // printf("%d", tamchecker);
    return !tamchecker;
}

bool IsReal(char str[])
{
    int tam = strlen(str);
    int tamchecker = strlen(str);
    int virgula = 0;
    for (int i = 0; i < tam; i++)
    {
        if ((str[i] >= '0') && (str[i] <= '9'))
        {
            tamchecker -= 1;
        }
        else if ((str[i] == '.') || (str[i] == ','))
        {
            tamchecker -= 1;
            virgula += 1;
        }
        if (virgula >= 2)
        {
            return false;
        }
    }
    // tamchecker = tamchecker - virgula;
    // printf("%d", tamchecker);
    return !tamchecker;
}

bool isFim(char s[])
{
    return ((strlen(s) >= 3) && (s[0] == 'F') && (s[1] == 'I') && (s[2] == 'M'));
}

int main()
{
    char string[1000];

    scanf(" %[^\n]", string);

    while (isFim(string) == false)
    {
        if (IsVogal(string) == true)
        {
            printf("SIM ");
        }
        else
        {
            printf("NAO ");
        }
        //------------------------------------------//
        if (IsConsoante(string) == true)
        {
            printf("SIM ");
        }
        else
        {
            printf("NAO ");
        }
        //------------------------------------------//
        if (IsInteiro(string) == true)
        {
            printf("SIM ");
        }
        else
        {
            printf("NAO ");
        }
        //------------------------------------------//
        if (IsReal(string) == true)
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        //------------------------------------------//

        scanf(" %[^\n]", string);
    }

    return 0;
}

File: Trabalhos Praticos/tp01/entrada e saida/TP01Q06 - Is em Java/Isthing.java
class Isthing {

    public static boolean isFim(String s) {
        return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    // ------------------------------------------//

    public static boolean IsVogal(String frase) {
        int tam = frase.length();
        int tamchecker = frase.length();
        for (int i = 0; i < tam; i++) {
            if ((frase.charAt(i) == 'a' || frase.charAt(i) == 'A') ||
                    (frase.charAt(i) == 'e' || frase.charAt(i) == 'E') ||
                    (frase.charAt(i) == 'i' || frase.charAt(i) == 'I') ||
                    (frase.charAt(i) == 'o' || frase.charAt(i) == 'O') ||
                    (frase.charAt(i) == 'u' || frase.charAt(i) == 'U') ||
                    frase.charAt(i) == ' ') {
                tamchecker -= 1;
            }
        }
        if (tamchecker == 0) {
            return true;
        }
        return false;
        // return tamchecker == 0 ? true : false;
    }

    // ------------------------------------------//

    public static boolean IsConsoante(String frase) {
        int tam = frase.length();
        int tamchecker = 0;
        for (int i = 0; i < tam; i++) {
            // Verifica se é uma letra do alfabeto
            if ((frase.charAt(i) >= 'a' && frase.charAt(i) <= 'z')
                    || (frase.charAt(i) >= 'A' && frase.charAt(i) <= 'Z')) {
                // Verifica se é uma vogal
                if (frase.charAt(i) == 'a' || frase.charAt(i) == 'A' ||
                        frase.charAt(i) == 'e' || frase.charAt(i) == 'E' ||
                        frase.charAt(i) == 'i' || frase.charAt(i) == 'I' ||
                        frase.charAt(i) == 'o' || frase.charAt(i) == 'O' ||
                        frase.charAt(i) == 'u' || frase.charAt(i) == 'U') {
                    tamchecker--; // Decrementa se for uma vogal
                }
            } else {
                tamchecker--; // Decrementa se não for uma letra do alfabeto
            }
        }
        if (tamchecker == 0) {
            return true;
        }
        return false;
    }

    // ------------------------------------------//

    public static boolean IsInteiro(String frase) {
        int tam = frase.length();
        int tamchecker = frase.length();
        for (int i = 0; i < tam; i++) {
            if (frase.charAt(i) >= '0' && frase.charAt(i) <= '9') {
                tamchecker -= 1;
            }
        }
        if (tamchecker == 0) {
            return true;
        }
        return false;
    }

    // ------------------------------------------//

    public static boolean IsReal(String frase) {
        int tam = frase.length();
        int tamchecker = frase.length();
        int virgula = 0;
        for (int i = 0; i < tam; i++) {
            if ((frase.charAt(i) >= '0') && (frase.charAt(i) <= '9')) {
                tamchecker -= 1;
            } else if ((frase.charAt(i) == '.') || (frase.charAt(i) == ',')) {
                tamchecker -= 1;
                virgula += 1;
            }
            if (virgula >= 2) {
                return false;
            }
        }
        if (tamchecker == 0) {
            return true;
        }
        return false;
    }

    // ------------------------------------------//

    public static void main(String[] args) {
        String str;

        str = MyIO.readLine();
        while (isFim(str) == false) {
            if (IsVogal(str) == true) {
                System.out.print("SIM ");
            } else {
                System.out.print("NAO ");
            }
            // ------------------------------------------//
            if (IsConsoante(str) == true) {
                System.out.print("SIM ");
            } else {
                System.out.print("NAO ");
            }
            // ------------------------------------------//
            
              if (IsInteiro(str) == true)
              {
              System.out.print("SIM ");
              }
              else
              {
              System.out.print("NAO ");
              }
              //------------------------------------------//
              if (IsReal(str) == true)
              {
              System.out.println("SIM ");
              }
              else
              {
              System.out.println("NAO ");
              }
              //------------------------------------------//             

            str = MyIO.readLine();
        }

    }
}